File: Find Missing And Repeating - GFG/find-missing-and-repeating.java
// { Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    // Driver code
    public static void main(String[] args) throws Exception {
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());
        while (t-- > 0) {
            int n = Integer.parseInt(br.readLine().trim());
            String[] str = br.readLine().split(" ");

            int[] a = new int[n];
            for (int i = 0; i < n; i++) {
                a[i] = Integer.parseInt(str[i]);
            }

            int[] ans = new Solve().findTwoElement(a, n);
            System.out.println(ans[0] + " " + ans[1]);
        }
    }
}// } Driver Code Ends


// User function Template for Java

class Solve {
       int[] findTwoElement(int arr[], int n) {
     		int xxory=0;
	        for(int i=0;i<n;i++)
	            xxory^=(i+1)^arr[i];
	        
	       int rmsb=xxory&-xxory;
	   
	       int x=0;
	       int y=0;
	       for(int i=0;i<n;i++) {
	    	   if((rmsb & arr[i])==0) x^=arr[i];
	    	   else y^=arr[i];
	       } 
	       
	       for(int i=1;i<=n;i++) {
	    	   if((rmsb&i)==0) x^=i;
	    	   else y^=i;
	       }
	       
	   int ans[]=new int[2];
	   for(int i=0;i<n;i++){
	       if(arr[i]==x){
	           ans[0]=x;
	           break;
	       }else if(arr[i]==y){
	           ans[0]=y;
	           break;
	       }
	   } 
	   
	   ans[1]=x^y^ans[0];
       return ans;
    }
}

File: Find Missing And Repeating - GFG/find-missing-and-repeating.cpp
// { Driver Code Starts
#include <bits/stdc++.h>

using namespace std;

 // } Driver Code Ends
class Solution{
public:
    int *findTwoElement(int *arr, int n) {
        // code here
        int *ans = new int[2]; 
        int sumA=0,c[n];
        int sumB=0;
        for(int i=0;i<n;i++){
            sumA+=arr[i];
            c[i]=arr[i];
            sumB+=(i+1);
        }
        sort(c,c+n);
        int i;
        for(i=1;i<n;i++){
            if(c[i]==c[i-1]) break;
        }
        ans[0]=c[i];
        ans[1]=sumB-sumA+ans[0];
        
        return ans;
        
    }
};

// { Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        int a[n];
        for (int i = 0; i < n; i++) {
            cin >> a[i];
        }
        Solution ob;
        auto ans = ob.findTwoElement(a, n);
        cout << ans[0] << " " << ans[1] << "\n";
    }
    return 0;
}  // } Driver Code Ends